novawallet/Modules/Gifts/GiftTransfer/GiftTransferConfirm/Factory/GiftTransferSubmissionFactory/Substrate/GiftTransferSubmissionFactory.swift

```diff
@@ -0,0 +1,141 @@
+import BigInt
+
+final class GiftTransferSubmissionFactory: GiftTransferSubmitting {
+    let giftFactory: GiftOperationFactoryProtocol
+    let giftsRepository: AnyDataProviderRepository<GiftModel>
+    let signingWrapper: SigningWrapperProtocol
+    let persistExtrinsicService: PersistentExtrinsicServiceProtocol
+    let persistenceFilter: ExtrinsicPersistenceFilterProtocol
+    let eventCenter: EventCenterProtocol
+    let chain: ChainModel
+    let asset: AssetModel
+    let selectedAccount: ChainAccountResponse
+    let extrinsicService: ExtrinsicServiceProtocol
+    let transferCommandFactory: SubstrateTransferCommandFactory
+    let operationQueue: OperationQueue
+
+    init(
+        giftFactory: GiftOperationFactoryProtocol,
+        giftsRepository: AnyDataProviderRepository<GiftModel>,
+        signingWrapper: SigningWrapperProtocol,
+        persistExtrinsicService: PersistentExtrinsicServiceProtocol,
+        persistenceFilter: ExtrinsicPersistenceFilterProtocol,
+        eventCenter: EventCenterProtocol,
+        chain: ChainModel,
+        asset: AssetModel,
+        selectedAccount: ChainAccountResponse,
+        extrinsicService: ExtrinsicServiceProtocol,
+        transferCommandFactory: SubstrateTransferCommandFactory,
+        operationQueue: OperationQueue
+    ) {
+        self.giftFactory = giftFactory
+        self.giftsRepository = giftsRepository
+        self.signingWrapper = signingWrapper
+        self.persistExtrinsicService = persistExtrinsicService
+        self.persistenceFilter = persistenceFilter
+        self.eventCenter = eventCenter
+        self.chain = chain
+        self.asset = asset
+        self.selectedAccount = selectedAccount
+        self.extrinsicService = extrinsicService
+        self.transferCommandFactory = transferCommandFactory
+        self.operationQueue = operationQueue
+    }
+
+    func createSubmitOperation(
+        dependingOn giftOperation: BaseOperation<GiftModel>,
+        amount: OnChainTransferAmount<BigUInt>,
+        assetStorageInfo: AssetStorageInfo?
+    ) -> BaseOperation<SubmittedGiftTransactionMetadata> {
+        AsyncClosureOperation { [weak self] completion in
+            guard let self else { throw BaseOperationError.parentOperationCancelled }
+
+            let gift = try giftOperation.extractNoCancellableResultData()
+
+            var callCodingPath: CallCodingPath?
+
+            let extrinsicClosure: ExtrinsicBuilderClosure = { [weak self] builder in
+                guard let self else { throw BaseOperationError.parentOperationCancelled }
+
+                let (newBuilder, codingPath) = try self.addingTransferCommand(
+                    to: builder,
+                    amount: amount,
+                    recepient: gift.giftAccountId,
+                    assetStorageInfo: assetStorageInfo
+                )
+
+                callCodingPath = codingPath
+
+                return newBuilder
+            }
+
+            extrinsicService.submit(
```

Comment:
The ExtrinsicSubmissionMonitor better suits here to get onchain execution guarantees